geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -598,7 +599,9 @@
       StatisticsFactory statisticsFactory =
           internalCache.getInternalDistributedSystem().getStatisticsManager();
       if (isGatewayReceiver()) {
-        stats = GatewayReceiverStats.createGatewayReceiverStats(statisticsFactory, sockName);
+        MeterRegistry meterRegistry = internalCache.getMeterRegistry();
+        stats = GatewayReceiverStats.createGatewayReceiverStats(statisticsFactory, sockName,
+            meterRegistry);
```

Comment:
Considering that the `GatewayReceiverStats` class already knows how the instrumentation should be done, have you though about passing the `InternalCache` as a parameter to the `createGatewayReceiverStats` method instead of `MeterRegistry`?.
I'm just worried about introducing a new instrumentation dependency within the class itself here, but maybe I'm missing something.

Reply:
We thought about it. Passing an internal cache would make `GatewayReceiverStats` depend not only on the registry, but also on `InternalCache` (which it has no other need for).

Further, it couples `GatewayReceiverStats` to the knowledge that an `InternalCache` is the source registry—knowledge which it otherwise does not need.

Finally, it would add complexity to the `GatewayReceiverStats` tests, which would have to set up an `InternalCache` that is not involved in any way in `GatewayReceiverStats` responsibilities.

To your concern: Yes, our design choice does introduce a new dependency into `AccessorImpl`. I would prefer to fix that in some other way. Perhaps by injecting a `GatewayReceiverStats` factory into `AccessorImpl`. That factory would encapsulate the dependency on the registry, and so `AccessorImpl` would not have to care.

We chose not to fix that in this PR, in order to remain focused on the task at hand (instrumenting this particular metric).